vm.go

```diff
@@ -4568,25 +4568,23 @@ func (_typeof) exec(vm *vm) {
 	case valueNull:
 		r = stringObjectC
 	case *Object:
-	repeat:
 		if v == nil {
 			r = stringFunction
 			vm.stack[vm.sp-1] = r
 			vm.pc++
 			break
 		}
 		switch s := v.self.(type) {
-		case *classFuncObject, *methodFuncObject, *funcObject, *nativeFuncObject, *wrappedFuncObject, *boundFuncObject, *arrowFuncObject:
+		// todo making a guess that templatedFuncObject is needed here
+		// todo perhaps use the same v.self.typeOf() call that is used in upstream
+		case *classFuncObject, *methodFuncObject, *funcObject, *nativeFuncObject, *wrappedFuncObject, *boundFuncObject, *arrowFuncObject, *templatedFuncObject:
```

Comment:
templatedFuncObject is needed here unless we move forward with more closely matching upstream.